### ISSUE TITLE
test arkenfox.js

### ISSUE THREAD

**Issue by arthuredelstein** [arthuredelstein/privacytests.org#80]

**Comment by Thorin-Oakenpants** [comment]
FYI - https://news.ycombinator.com/item?id=29850780

Interesting that you would want to test AF. I already know what it would be. If you want to do it correctly ... it's arkenfox with uBO + [two filters added](https://github.com/arkenfox/user.js/wiki/4.1-Extensions#small_orange_diamond-extensions) :) .. and I only just switched if from FPI to dFPI (strict) when v96 is final

It should pass (here's my prediction)
- all state except blobs same as LW
- all navigation (we block all referrers by default and recommend an extension to whitelist origin-destinations)
- all HTTPS
- misc is all n/a (GPC is default off, there is no IP anonymizing)
- all fingerprinting
- all/most tracking query params (if you add those two uBO filters lists)
- all tracker content blocking (with uBO)

It's slightly hardened, and not everyone can live with it like that, mostly referrers

We gonna need a bigger set of tests :) What do you suggest ?

**Comment by Thorin-Oakenpants** [comment]
just to be clear, I think adding AF is a waste of time, but feel free to do a one off test for me :)

> We gonna need a bigger set of tests :) What do you suggest ?

I have some ideas
- under navigation test? beacon, link-pre-fetching, predictor
- sanitizing: e.g. cache/HSTS etc, cookies + site data, etc
   - will look impressive for AF/LW and of course TB (being in PB Mode)
- a "security" section: e.g. the five safebrowsing types

**Comment by arthuredelstein** [comment]
> just to be clear, I think adding AF is a waste of time, but feel free to do a one off test for me :)

Why do you think it is a waste of time? I was thinking it would be an interesting thing to show alongside various extensions.

>     * under navigation test? beacon, link-pre-fetching, predictor
> 
>     * sanitizing: e.g. cache/HSTS etc, cookies + site data, etc
> 
>     * a "security" section: e.g. the five safebrowsing types

These are good ideas. Thanks!

**Comment by Thorin-Oakenpants** [comment]
> These are good ideas. Thanks!

I just cherry picked a few things from AF's prefs that I know it will pass and many others won't 😀 

> Why do you think it is a waste of time?

Because we basically already know the answers - see my prediction. In the current tests, AF96 is almost (95%) the same as LW (I made LW have an intervention for their own health and then spent the last 8 months helping one of their devs with info and suggestions). About the only diff is referrers in the current tests.

If you add the things that AF will pass, it would make me look like a genius - and this includes more diffs from LW (but probably not for long if they would only listen to me on the last few items). So eventually, you're just going to end up with LW and an extra tick or two

Sure it's interesting. IDK how you want to frame it. A new set of "custom" browser results makes sense i.e not a fork and not  default. IMO, it should be

arkenfox with uBO + 2 filterlists will become immaterial
      - `privacy.query_stripping.enabled` will eventually cover query parameter tests
      - ETP Strict already covers TP
      - both are forms of **_enumerating badness_** with small test sets (of the biggest culprits), so I would be weary of grading it and giving it scores other than a binary yes/no and an apt description

and arkenfox on it's own is not an "extension" or a fork. IDK how/where you would fit it in

Up to you :) I'm more concerned with how others read the results - because without uBO + extras, it's not what I suggest. But then the AF user.js is a "template" that flips 150 prefs (and essentially warns against the rest), of which about 10 prefs (highlighted and suggested to change if needed) cover 99% of the breakage, and users are going to change some of those. So it's not real world.

**Comment by d4rklynk** [comment]
> Up to you :) I'm more concerned with how others read the results - because without uBO + extras, _it's not what I suggest_. On the other hand the AF user.js is a "template" that flips 150 prefs (and essentially warns against the rest), of which about 10 prefs (highlighted and suggested to change if needed) cover 99% of the breakage, and users are going to change some of those. So it's not real world: there is no default

But in LW there are some settings you can flip too. I think adding AF is a really good idea. You remove the top 1 issue of AF (I think it's 1601) and you good to go. 
I don't think you should test any browser with any add-on though.
**Just test AF with the 1601 fix and that's all**

**Comment by Thorin-Oakenpants** [comment]
from https://github.com/privacytests/privacytests.org/issues/80#issuecomment-1008282746
> I have some ideas
> - sanitizing: e.g. cache/HSTS etc, cookies + site data, etc
>    - will look impressive for AF/LW and of course TB (being in PB Mode)

I can't believe it's only been 18 months, time flies ... how did my prediction stack up? :) thanks Arthur for the tests

![woo](https://github.com/privacytests/privacytests.org/assets/16656956/bbbd6372-f590-4442-b32f-267ca0b9ed77)

**Comment by arthuredelstein** [comment]
> from [#80 (comment)](https://github.com/privacytests/privacytests.org/issues/80#issuecomment-1008282746)
> 
> > I have some ideas
> > 
> > * sanitizing: e.g. cache/HSTS etc, cookies + site data, etc
> >   
> >   * will look impressive for AF/LW and of course TB (being in PB Mode)
> 
> I can't believe it's only been 18 months, time flies ... how did my prediction stack up? :) thanks Arthur for the tests
> 
> ![woo](https://user-images.githubusercontent.com/16656956/252507080-bbbd6372-f590-4442-b32f-267ca0b9ed77.png)



> from [#80 (comment)](https://github.com/privacytests/privacytests.org/issues/80#issuecomment-1008282746)
> 
> > I have some ideas
> > 
> > * sanitizing: e.g. cache/HSTS etc, cookies + site data, etc
> >   
> >   * will look impressive for AF/LW and of course TB (being in PB Mode)
> 
> I can't believe it's only been 18 months, time flies ... how did my prediction stack up? :) 

Perfect! :) And, yeah, I can't believe it's been 18 months either...

**Comment by Thorin-Oakenpants** [comment]
OT: hmm, I noticed FF PB mode failed cross-session 3p prefetch cache - did you want to alert someone at Mozilla (or do you want me to try to do it), or is it a bug

**Comment by Thorin-Oakenpants** [comment]
no thanks, please don't highlight/promote arkenfox via privacytests - I'm trying to kill it

**Comment by Thorin-Oakenpants** [comment]
https://github.com/mullvad/mullvad-browser/issues/1#issuecomment-1495998714
- > PS: long term I am hoping we can retire arkenfox ... and just use MB - but they're just a little too divergent for now

https://github.com/arkenfox/user.js/issues/1842#issuecomment-2153421218
- > ... So for the last 8 months we (MB/rui/me) have been talking about how (to kill arkenfox and) improve MB (and TB) ... start them in normal mode

basically this comes down to getting TB and MB to relax disk usage when the user opts into it - basically I think both should start in persistent mode but sanitize, but there's a lot of work to do before we get to that
